### PR TITLE
chore(clients/go): OAuth credentials cache

### LINKED ISSUE #3063: OAuth credential caching
**Is your feature request related to a problem? Please describe.**

Multiple clients on the same node/application will most likely use the same client secret/ID, and will therefore be pulling the same OAuth credentials. To decrease traffic and avoid potential rate limits, caching those credentials so they are shareable in the same context (e.g. node) would be useful.

**Describe the solution you'd like**

Add a credentials cache as a YAML file on the local filesystem, default path `$HOME/.camunda/credentials` with the following format:

```yaml
---
audience:
  auth:
    credentials:
      access_token: "Token", # access token
      expires_in: 3600, # expiry in seconds
      token_type: "Bearer", # token type, most likely always Bearer
      scope: "grpc" # Optional scope
audience2: # ... more audiences
```

**Describe alternatives you've considered**

Users concerned with caching/rate limiting should authenticate through a proxy and control it themselves.

**Additional context**

Should be similar to the Java client implementation

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

Implements an on disk credentials cache for the Go client following the format delineated by the cloud team:

```yaml
---
audience:
  auth:
    credentials:
      access_token: "Token", # access token
      expires_in: 3600, # expiry in seconds
      token_type: "Bearer", # token type, most likely always Bearer
      scope: "grpc" # Optional scope
audience2: # ... more audiences
```

Enforces that the OAuthCredentialsProvider must be used with a cache. Default path to the cache `.camunda/credentials` relative to the user's home directory, but can be overwritten by the environment variable `ZEEBE_CLIENT_CACHE_PATH`.

## Related issues

closes #3063 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Comment by menski** [comment]
the test `TestOAuthProviderRetry` is failing on Jenkins and when I run all test locally using `make test`

**Comment by npepinpe** [comment]
I added a test last minute to test the env var, and of course I forgot to reset it so the tests running after were using the env var variable 🙈

**Comment by npepinpe** [comment]
Still needs to be rebased once the others PR are merged though

File: clients/go/zbc/oauthCredentialsCache_test.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+	err = os.Setenv(OAuthCachePathEnvVar, file.Name())
+	defer os.Unsetenv(OAuthCachePathEnvVar)
+	cache, err := NewOAuthYamlCredentialsCache(fakePath)
+	require.NoError(t, err)
+	require.Equal(t, file.Name(), cache.path)
+	require.Empty(t, cache.audiences)
+}
+
+func copyCredentialsCacheGoldenFileToTempFile() string {
```
Comment (npepinpe) [inline clients/go/zbc/oauthCredentialsCache_test.go]: that sweet error handling 😎

**Review by menski — APPROVED** [review]
bors r+

**Pull request merged**